pkg/api/types.go

```diff
@@ -983,6 +994,16 @@ type Binding struct {
 	Target ObjectReference `json:"target"`
 }
 
+// DeleteOptions may be provided when deleting an API object
+type DeleteOptions struct {
+	TypeMeta `json:",inline"`
+
+	// Optional duration in seconds before the object should be deleted. Value must be non-negative integer.
+	// The value zero indicates delete immediately. If this value is nil, the default grace period for the
+	// specified type will be used.
+	GracePeriodSeconds *int64 `json:"gracePeriodSeconds"`
```

Comment:
This is a pointer so we can distinguish Go defaulting from the user specifying 0?